gateway-test/src/test/java/org/apache/knox/gateway/AmbariServiceDefinitionTest.java

```diff
@@ -352,4 +366,28 @@ public void contextPathInViewsResponse() throws Exception {
     LOG_EXIT();
   }
 
+  private static void createTestKeystore()
```

Comment:
Duplicated in gatewaytestdriver?

Reply:
It is.. I had issue with that too, but I was not sure if introducing the GatewayDriver was appropriate.  If we think it is ok, then I am all for the code reuse.

Reply:
Hmmmm - Not sure if it is possible to move the create test keystore method from `GatewayTestDriver` to `gateway-test-utils` module `TestUtils`? It looks like it relies on stuff in `gateway-server` like `X509CertificateUtil`. 

Just seems really ugly to have this block of code copied around.

Reply:
Actually - `X509CertificateUtil` is in `gateway-spi` so shouldn't be too bad to move to `TestUtils`. This should avoid needing an additional dependency in most cases. Much easier to add a test dependency on `gateway-test-utils` than on `gateway-test-release-utils`.

Reply:
Adding gateway-spi as a dependency of gateway-test-utils creates a cycle...
```
[ERROR] The projects in the reactor contain a cyclic reference: Edge between 'Vertex{label='org.apache.knox:gateway-spi:1.3.0-SNAPSHOT'}' and 'Vertex{label='org.apache.knox:gateway-i18n:1.3.0-SNAPSHOT'}' introduces to cycle in the graph org.apache.knox:gateway-i18n:1.3.0-SNAPSHOT --> org.apache.knox:gateway-test-utils:1.3.0-SNAPSHOT --> org.apache.knox:gateway-spi:1.3.0-SNAPSHOT --> org.apache.knox:gateway-i18n:1.3.0-SNAPSHOT -> [Help 1]
```

Reply:
I will copy the `X509CertificateUtil` into the project to make things simple.